eng/pipelines/templates/steps/cmake-build.yml

```diff
@@ -3,7 +3,7 @@ parameters:
   BuildType: Debug
 
 steps:
-  - script: mkdir build
+  - script: mkdir -p build
```

Comment:
why?

Reply:
See https://github.com/Azure/azure-sdk-for-c/issues/1122 and https://github.com/Azure/azure-sdk-for-c/pull/1100#issuecomment-679399544